### PR TITLE
Added documentation examples for `nonlocal-and-global`.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Refs https://github.com/PyCQA/pylint/issues/5953

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
I'm not sure about the "good" example because I don't know what the problem being solved is, could you explain @mbyrnepr2 ? Maybe not using global nor non-local works ?

**Comment by mbyrnepr2** [comment]
> I'm not sure about the "good" example because I don't know what the problem being solved is, could you explain @mbyrnepr2 ? Maybe not using global nor non-local works ?

The idea was to create a closure to remember the last value and update the global one.
However, since you need to ask then I think the example isn't clear. OK with me to close this

File: doc/data/messages/n/nonlocal-and-global/bad.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+NUMBER = 42
+
+
+def update_number(number):  # [nonlocal-and-global]
+    global NUMBER
+    nonlocal NUMBER
+    NUMBER = number
+    print(f"New global number is: {NUMBER}")
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/n/nonlocal-and-global/bad.py:8]:
```suggestion
    print(f"New global number is: {NUMBER}")
    

update_number(24)
```

File: doc/data/messages/n/nonlocal-and-global/good.py [hunk 2]
```diff
@@ -1 +1,19 @@
...
+        print(f"New global number is: {NUMBER}")
+
+        nonlocal previous
+        print(f"Previous global number was: {previous}")
+        previous = number
+    return update_number
+
+
+update_number = create_number_updater()
+update_number(24)
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/n/nonlocal-and-global/good.py:19]:
```suggestion
NUMBER = 42
NUMBER = 24
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
No no, it's just that I don't understand the example/message and what the intent leading to the error was. Your example in bad.py is illustrating the problem well. Would this suggestion work for example ?

**Comment by mbyrnepr2** [comment]
> No no, it's just that I don't understand the example/message and what the intent leading to the error was. Your example in bad.py is illustrating the problem well. Would this suggestion work for example ?

I see what you mean. My way of thinking of it was to somehow include usage of both global & nonlocal in the good example also. But that probably isn't necessary. Thanks for simplifying it!

File: doc/data/messages/n/nonlocal-and-global/good.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply doc/data/messages/n/nonlocal-and-global/good.py]: @DanielNoord, do you mind giving your opinion on this ?

Comment (DanielNoord) [reply doc/data/messages/n/nonlocal-and-global/good.py]: Shouldn't the good example use either `global` or `nonlocal`? The warning is about using both and you needing to decide which it should not be, not remove both of them.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**